# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1245741430 -7200
#      Tue Jun 23 09:17:10 2009 +0200
# Node ID bb0697c70c1617850d5655fb15cd2c15236e3e77
# Parent  610bf90fce2a2e2db0b268addaac3c8638b99d0d
simplify tree_evaluator::visit_try_catch_command

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2009-06-23  Jaroslav Hajek  <highegg@gmail.com>
+
+	* pt-eval.cc (tree_evaluator::visit_try_catch_command): Simplify.
+
 2009-06-23  Jaroslav Hajek  <highegg@gmail.com>
 
 	* DLD-FUNCTIONS/cellfun.cc: Modernize unwind_protect usage.
 	* DLD-FUNCTIONS/daspk.cc: Ditto.
 	* DLD-FUNCTIONS/dasrt.cc: Ditto.
 	* DLD-FUNCTIONS/dassl.cc: Ditto.
 	* DLD-FUNCTIONS/eigs.cc: Ditto.
 	* DLD-FUNCTIONS/lsode.cc: Ditto.
diff --git a/src/pt-eval.cc b/src/pt-eval.cc
--- a/src/pt-eval.cc
+++ b/src/pt-eval.cc
@@ -888,23 +888,17 @@ tree_evaluator::visit_try_catch_command 
       error_state = 0;
 
       // Unwind stack elements must be cleared or run in the reverse
       // order in which they were added to the stack.
 
       // For clearing the do_catch_code cleanup function.
       unwind_protect::discard ();
 
-      // For restoring Vdebug_on_warning, Vdebug_on_error, and
-      // buffer_error_messages.
-      unwind_protect::run ();
-      unwind_protect::run ();
-      unwind_protect::run ();
-
-      // Also clear the frame marker.
+      // Run the rest of the frame.
       unwind_protect::run_frame (uwp_frame);
     }
 }
 
 static void
 do_unwind_protect_cleanup_code (void *ptr)
 {
   tree_statement_list *list = static_cast<tree_statement_list *> (ptr);
